server/portal/apps/projects/unit_test.py

```diff
@@ -3,208 +3,1091 @@
+            "created": "2023-03-07T19:31:17.292220Z",
+            "projectId": "test.project-2",
+            "members": [
+                {"user": ws_o.get_project_user("username"), "access": "owner"},
+                {"user": ws_o.get_project_user("new_user"), "access": "reader"},
+            ],
+        }
+        project_post_update = mock_get_project(client, workspace_id)
+        assert project_post_update["members"] == [
+            {"user": ws_o.get_project_user("username"), "access": "owner"},
+            {"user": ws_o.get_project_user("new_user"), "access": "reader"},
+        ]
 
 
-@pytest.mark.skip(reason="TODOv3: update with new Shared Workspaces operations")
-def test_add_pi_unauthorized(mock_owner, django_user_model, mock_tapis_client, mock_storage_system, project_model, mock_signal, mock_service_account):
-    'Test add member.'
+# Remove a user from a project
+def test_remove_user(mock_tapis_client, mock_owner, authenticated_user):
+    with patch_workspace_operations() as mocks:
+        mock_service_account = mocks[0]
+        mock_create_shared_workspace = mocks[2]
+        mock_increment_workspace_count = mocks[3]
+        mock_create_workspace_dir = mocks[4]
+        mock_add_user_to_workspace = mocks[6]
+        mock_get_project = mocks[7]
+        mock_get_workspace_role = mocks[8]
+        client = mock_tapis_client
+        mock_service_account.return_value = mock_tapis_client
+        title = "Test Workspace 1"
+        description = ""
+        # Create first project
+        workspace_number = 2
 
-    prj = project_model.create(mock_tapis_client, 'Test Title', 'PRJ-123', mock_owner)
-    prj.storage.roles.for_user.return_value = MagicMock(role='USER', ADMIN='ADMIN')
-    assert not prj._can_edit_member(mock_owner)
+        # Creates an initial shared workspace with two users already defined
+        workspace_id = create_shared_workspace_2_user(
+            client,
+            title,
+            description,
+            mock_service_account,
+            mock_owner,
+            mock_create_shared_workspace,
+            mock_increment_workspace_count,
+            mock_create_workspace_dir,
+            mock_add_user_to_workspace,
+            mock_get_project,
+            mock_get_workspace_role,
+            authenticated_user,
+            workspace_num=workspace_number,
+        )
+        # This change_user_role never returns a get project, only uses Tapis calls
+        # Mocking what the contents of the project would be like after action
+        project_pre_update = mock_get_project(client, workspace_id)
+        assert project_pre_update["members"] == [
+            {"user": ws_o.get_project_user("username"), "access": "owner"},
+            {"user": ws_o.get_project_user("new_user"), "access": "writer"},
+        ]
+        ws_o.remove_user(client, workspace_id, "new_user")
+        # TODO: Fix the assertions and potential return value
```

Comment:
Can you elaborate on what this means?

Reply:
Duplicates based on https://github.com/TACC/Core-Portal/pull/987#discussion_r1828492022 some mock tapis calls are not executing correctly. Would love to have a discussion on this if these assertions like in https://github.com/TACC/Core-Portal/pull/987#discussion_r1828492022 should be included or removed